### ISSUE TITLE
Make `trace` leave warning in the codes.

### ISSUE THREAD

**Issue by yairchu** [snoyberg/mono-traversable#136]

**Comment by snoyberg** [comment]
Unless I'm mistaken, this would involve defining a new version of the trace
identifier, which would now conflict with Debug.Trace. This would be a
breaking change, but I'm OK with the idea (we do this already for
`undefined`). Am I mistaken about my assumptions on this? Otherwise,
interested in sending a PR?

**Comment by yairchu** [comment]
Yes, this would define a new `trace` function which is a proxy for `Debug.Trace.trace`. iiuc that's fine so I'll proceed with a PR
